### ISSUE TITLE
FileOption extension for another language is compiled into python

### ISSUE THREAD

**Issue by samskiter** [google/protobuf#995]
I'm using a plugin for protobuf (this one: https://github.com/alexeyxo/protobuf-objc) and so my proto file has an import and an option set at the top of it:
```
import "google/protobuf/objectivec-descriptor.proto";
option (google.protobuf.objectivec_file_options).class_prefix = "ABC";
```

Should I still be able to compile this to python? The generated python has this import included:

```
import google.protobuf.objectivec_descriptor_pb2
```

I'm using this command to compile:
```
protoc --proto_path=../path/to/protobuf-objc --proto_path=. --python_out=. mymessages.proto
```

**Comment by xfxyjwf** [comment]
Yes, you should be able to compile that to python. Because your proto depends on objective-descriptor.proto, you will need to compile that proto to python as well.

**Comment by samskiter** [comment]
So including a file doesn't cause it to be compiled? I'l need the following?

```
protoc --proto_path=../path/to/protobuf-objc --proto_path=. --python_out=. mymessages.proto ../path/to/protobuf-objc/google/
```

**Comment by samskiter** [comment]
@xfxyjwf This doesn't work either I'm afraid. objectivec_descriptor_pb2.py is generated but I get resulting in the error when trying to use mymessages_pb2.py:
```
ImportError: No module named objectivec_descriptor_pb2
```

**Comment by xfxyjwf** [comment]
When you compile your python code, have you added the generated source of google/protobuf/objectivec_descriptor_pb2.py? You mentioned the file is generated, so I assume after updating your Python build scripts it should all work well.

**Comment by samskiter** [comment]
@xfxyjwf The generated mymessages_pb2.py file imports the objectivec_descriptor_pb2 file as follows:
```
import google.protobuf.objectivec_descriptor_pb2
```
And the file `google/protobuf/objectivec_descriptor_pb2.py` was generated and exists alongside mymessages_pb2.py, yes.

**Comment by samskiter** [comment]
@xfxyjwf I've found there is a conflict with the python protobuf module - it has a matching package of `google.protobuf`. I've got it a little closer to working by:
1. fiddling the namespacing by renaming the generated `google` folder to `google1`
2. adding an __init__.py file to both the generated `google1` and `protobuf` directories.

So,

1. If protoc is generating a package for me, should it not have added the __init__.py files for me?
2. Should the protobuf-objc plugin be sharing the google namespace?
